src/encoded/types/biosample.py

```diff
@@ -1210,3 +1235,18 @@ def construct_biosample_summary(phrases_dictionarys, sentence_parts):
     rep = dict((re.escape(k), v) for k, v in rep.items())
     pattern = re.compile("|".join(rep.keys()))
     return pattern.sub(lambda m: rep[re.escape(m.group(0))], sentence_to_return)
+
+
+def is_part_of(request, biosample_id, biosample_ontology, part_of_object, checked_biosamples):
+    if biosample_ontology != part_of_object['biosample_ontology']:
+        return biosample_id
+    elif 'part_of' not in part_of_object:
+        return part_of_object['@id']
+    else:
+        biosample_id = part_of_object['@id']
+        if biosample_id not in checked_biosamples:
+            checked_biosamples.append(biosample_id)
+            part_of_object = request.embed(part_of_object['part_of'], '@@object')
+            return is_part_of(request, biosample_id, biosample_ontology, part_of_object, checked_biosamples)
+        else:
+            return part_of_object['@id']
```

Comment:
I think here it should be `return biosample_id` rather than `return part_of_object['@id']`

Reply:
Oh yes, thank you will change that now